src/lib/utils/locking_allocator/locking_allocator.cpp

```diff
@@ -68,8 +70,32 @@ mlock_allocator::~mlock_allocator()
 
 mlock_allocator& mlock_allocator::instance()
    {
-   static mlock_allocator mlock;
-   return mlock;
+   return mlock_allocator_instance;
+   }
+
+/*
+Schwarz counter / nifty counter idiom
+see: https://en.wikibooks.org/wiki/More_C%2B%2B_Idioms/Nifty_Counter
+*/
+static int nifty_counter;
+static typename std::aligned_storage<sizeof (mlock_allocator), alignof (mlock_allocator)>::type
+   mlock_allocator_buf;
+mlock_allocator& mlock_allocator_instance = reinterpret_cast<mlock_allocator&> (mlock_allocator_buf);
```

Comment:
Can these all live in an anon namespace?

Reply:
The first two of them can, but `mlock_allocator_instance` needs to remain outside that namespace.